vstools/utils/ranges.py

```diff
@@ -74,20 +75,37 @@ def replace_ranges(
     if not mismatch:
         check_ref_clip(clip_a, clip_b)
 
+    nranges = normalize_ranges(clip_b, ranges)
+
+    if use_plugin and hasattr(vs.core, 'remap'):
+        try:
+            return vs.core.remap.ReplaceFramesSimple(
+                clip_a, clip_b, mismatch=mismatch,
+                mappings=' '.join(f'[{s} {e + (exclusive if s != e else 0)}]' for s, e in nranges)
+            )
+        except vs.Error as e:
+            msg = str(e).replace('vapoursynth.Error: ReplaceFramesSimple: ', '')
+
+            match msg:
+                case "Clip lengths don't match":
+                    raise LengthMismatchError(replace_ranges, reason=f'({clip_a.num_frames} != {clip_b.num_frames})')
```

Comment:
You should put this reason logic and formatting into the actual classes, and only pass the clips.